nixos/modules/programs/ydotool.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.programs.ydotool;
+in
+
+{
+  ###### interface
+
+  options = {
+    programs.ydotool = {
+      enable = lib.mkEnableOption (lib.mdDoc "ydotool, a generic Linux command-line automation tool");
+    };
+  };
+
+  ###### implementation
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.ydotool ];
+
+    systemd.user.services.ydotoold = {
+      description = "Starts ydotoold service";
+      wantedBy = [ "defaul.target" ];
```

Comment:
This should be multi-user.target

Reply:
`default.target` should be correct, `multi-user.target` is only for system services.

Reply:
Actually it hit me just now that this depends on my user being in the group `input`. Maybe this should be changed to a system wide service and be ran as root instead?

Reply:
As the ydotool client uses `XDG_RUNTIME_DIR` i didn't change it back to a system service, but added documentation about the user needing to be in the `input` group.